SimFastTiming/FastTimingCommon/interface/MTDDigitizer.h

```diff
@@ -165,7 +177,12 @@ namespace mtd_digitizer {
     
 
   template<class Traits>
-  void MTDDigitizer<Traits>::beginRun(const edm::EventSetup & es) {
+  void MTDDigitizer<Traits>::beginRun(const edm::EventSetup & es) {    
+    edm::ESHandle<MTDGeometry> geom;
+    if( geom_ == nullptr ) {
+      es.get<MTDDigiGeometryRecord>().get(geom);
+      geom_ = geom.product();
+    }
```

Comment:
In principle this pattern is not the best approach. What if `MTDDigiGeometryRecord` would have an IOV change within a job? Ok, in practice that is not likely to happen I suppose (for variety of reasons), but nevertheless it would be better to get the `MTDGeometry` from EventSetup unconditionally. The caching within EventSetup is performant-enough to allow getting ESProducts per event.

Reply:
OK, I've removed the check on the pointer